### ISSUE TITLE
Consider changing naming style

### ISSUE THREAD

**Issue by msimberg** [eth-cscs/DLA-Future#576]
TL;DR: It'd be nice to follow the standard library (and pika, etc.) naming style of `snake_case` for almost everything except template parameters (which are `CapitalCase`).

Trigger: With senders and sender adaptors the pattern of "callable objects" (CPO, niebloid, whatever you want to call them) is more common. With our current naming scheme a plain function is:
```
T myFunction(...);
```
whereas a callable object would be
```
struct MyFunction { T operator()(...); } my_function{};
```
which means at a callsite the user now sees inconsistent casing depending on the _implementation_ of a function, which they don't really need to care about. (Semi-related: I'd like to start getting rid of the `_o` suffix for the auto-generated callable objects as that is also an implementation detail).

Other reasons:
- personal preference
- consistency with pika and the standard library

I know that this is ultimately a subjective matter, but like #340 we shouldn't be afraid to make this change sooner rather than later while we can still do it without breaking users. Of course, if there's no consensus I suppose the status quo wins.

**Comment by rasolca** [comment]
I'm not against changing the function name style. However I would like to keep CapitalCase for types, classes and structs.
The reason is simply the possibility to write the following:
```
Tile<...> tile{...};
```
